mathesar_ui/src/pages/schemas/DeleteSchema.svelte

```diff
@@ -0,0 +1,77 @@
+<script lang="ts">
+  import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+  import {
+    Button,
+    Modal,
+    Icon,
+  } from '@mathesar-components';
+  import type { Schema } from '@mathesar/App.d';
+  import { States } from '@mathesar/utils/api';
+  import { deleteSchema } from '@mathesar/stores/schemas';
+  import { currentDBName } from '@mathesar/stores/databases';
+
+  export let isOpen = false;
+  export let schema: Schema;
+
+  let state: States = States.Idle;
+  let error;
+
+  async function removeSchema() {
+    if (schema) {
+      try {
+        state = States.Loading;
+        error = null;
+        await deleteSchema($currentDBName, schema.id);
+        state = States.Done;
+      } catch (err) {
+        state = States.Error;
+        error = (err as Error).message;
+      }
+    }
+  }
+</script>
+
+<Modal class="schema-modal" bind:isOpen closeOnEsc={state !== States.Loading}>
+  <div class="header">
+    Delete schema '{schema?.name}'
+  </div>
+
+  {#if schema?.has_dependencies}
+    <div class="help-text">
+      All objects in this schema will be deleted permanently, including tables and views.
```

Comment:
```suggestion
      All objects in this schema will be deleted permanently, including (but not limited to) tables and views. Some of these objects may not be visible in the Mathesar UI.
```

Reply:
@kgodey I've committed this suggestion. Thanks!